src/main/java/seedu/taassist/model/TaAssist.java

```diff
@@ -94,6 +107,32 @@ public void removeStudent(Student key) {
         students.remove(key);
     }
 
+    //// class-level operations
+
+    /**
+     * Returns true if a class with the same identity as {@code moduleClass} exists in TA-Assist.
+     */
+    public boolean hasModuleClass(ModuleClass moduleClass) {
+        requireNonNull(moduleClass);
+        return moduleClasses.contains(moduleClass);
+    }
+
+    /**
+     * Adds a class to TA-Assist.
+     * The class must not already exist in TA-Assist.
+     */
+    public void addModuleClass(ModuleClass c) {
+        moduleClasses.add(c);
+    }
+
+    /**
+     * Removes {@code key} from this {@code TaAssist}.
+     * {@code key} must exist in TA-Assist.
+     */
+    public void removeModuleClass(ModuleClass key) {
+        moduleClasses.remove(key);
```

Comment:
What does `key` refer to? Will `moduleClass` be better?

Reply:
Great suggestion!